### ISSUE TITLE
yaml options are ignored if too many empty new lines are added

### ISSUE THREAD

**Issue by crazyminecuber** [quarto-dev/quarto-cli#3560]
### Bug description

I expect this to fail because of unknown documentclass (or use the locally defined asdf), but it instead compiles with the default value for documentclass.
```
---
title: "Quarto Basics"
format:
  pdf: 
    documentclass: asdf


---

# Hi
foo
```

If I remove one or both of the empty lines before the ending, `---` I get the expected behavior.

Gives the expected result: not compling
```
---
title: "Quarto Basics"
format:
  pdf: 
    documentclass: asdf

---

# Hi
foo
```
same here
```
---
title: "Quarto Basics"
format:
  pdf: 
    documentclass: asdf
---

# Hi
foo
```

I consider this to be a bug that should be fixed. Either not compile at all if empty lines before `---` exists or comile and parse YAML correctly.

# Environment

OS: Fedora 37

Editor: VScode.

quarto --version

```
1.2.269
```

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by cscheid** [comment]
That's not a behavior we wish to control. It comes from pandoc, which requires that its YAML blocks be strictly bracketed (without newlines). It's admittedly an annoying issue; I believe it arises `---` having multiple interpretations in Pandoc. It needs a way to track which ones are `hr` elements and which ones are yaml block delimiters, and it uses surrounding whitespace. That causes this sort of feelsbad moments.

We plan to implement linters to warn about this in the medium-term future, but this is not behavior we intend to override.

**Comment by crazyminecuber** [comment]
Is this behaviour known to pandoc or should i report it to them? If they know about it, what the reasoning for it?

**Comment by kapsner** [comment]
We have noticed (and described [here](https://github.com/kapsner/authors-block/issues/2#issuecomment-1356239332)) that also introducing >1 empty lines somewhere within the YAML-header of a document seems also to lead to this issue and the document header is not rendered.

**Comment by cscheid** [comment]
@kapsner Yes, that's likely the same issue. I agree with you that It's unfortunate syntax, but we're not going to deviate from Pandoc's syntax in this case; it would cause more harm than good.

**Comment by kapsner** [comment]
Yes, I totally understand. I think that adding a linter-option, as mentioned above, would be sufficient.
